### PR TITLE
Add Thorchain

### PR DESCRIPTION
**Pull request by Ehsan-saradar**

### CODE REVIEW

File: utils/thorchain-tx-provider.ts [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+            isDeposit: false,
+          } as SpecificThorchain;
+
+          resolve(specificThorchain);
+        });
+      });
+    });
+  };
+
+  public getGasLimit = (): number => {
```
Comment (johnnyluo) [inline utils/thorchain-tx-provider.ts:126]: thorchain doesn't have gas limit

File: utils/thorchain-tx-provider.ts [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+  public getEstimateTransactionFee = (
+    cmcId: number,
+    currency: Currency
+  ): Promise<string> => {
+    return new Promise((resolve) => {
+      api
+        .cryptoCurrency(cmcId, currency)
+        .then((price) => {
+          const gwei = formatUnits(
+            (this.gasPrice + this.maxPriorityFeePerGas) *
```
Comment (johnnyluo) [inline utils/thorchain-tx-provider.ts:93]: this is EVM only , not thorchain

File: utils/thorchain-tx-provider.ts [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+          (chain) => chain.name === transaction.chain.name
+        ).derivationKey,
+        isNativeToken: true,
+        logo: transaction.chain.ticker.toLowerCase(),
+      });
+      this.getSpecificTransactionInfo(coin).then((specificData) => {
+        const thorchainSpecific = create(THORChainSpecificSchema, {
+          accountNumber: BigInt(specificData.accountNumber),
+          fee: BigInt(specificData.fee),
+          isDeposit: false,
```
Comment (johnnyluo) [inline utils/thorchain-tx-provider.ts:151]: isDeposit can get from specifData as well

File: utils/constants.ts [hunk 4]
```diff
@@ -120,7 +127,61 @@ export enum RequestMethod {
   WALLET_WATCH_ASSET = "wallet_watchAsset",
   WEB3_CLIENT_VERSION = "web3_clientVersion",
 }
-
+export enum ThorRequestMethod {
+  ETH_ACCOUNTS = "eth_accounts",
```
[the remark this answers is not in the document]
Comment (Ehsan-saradar) [reply utils/constants.ts]: Resolved

File: utils/thorchain-tx-provider.ts (hunk 1 above)
Comment (Ehsan-saradar) [reply utils/thorchain-tx-provider.ts]: Removed

File: utils/thorchain-tx-provider.ts (hunk 2 above)
Comment (Ehsan-saradar) [reply utils/thorchain-tx-provider.ts]: Fixed

File: utils/thorchain-tx-provider.ts (hunk 3 above)
Comment (Ehsan-saradar) [reply utils/thorchain-tx-provider.ts]: I will add deposit in another PR

**Review by johnnyluo — APPROVED** [review]
LGTM

**Pull request merged**